network/p2p/libP2PHostWrapper.go

```diff
@@ -0,0 +1,123 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	libp2pnet "github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	tptu "github.com/libp2p/go-libp2p-transport-upgrader"
+	"github.com/libp2p/go-libp2p/config"
+	"github.com/libp2p/go-tcp-transport"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/rs/zerolog"
+)
+
+// libP2PHostWrapper is an internal data structure to this package that encapsulates
+// all components that p2p package affiliates with a libp2p host.
+// Purpose of encapsulation is to ease passing the entire libp2p host affiliated components
+// around.
+type libP2PHostWrapper struct {
```

Comment:
the `libp2pNode.go` was the `libp2p` wrapper. Wouldn't it simplify the code if we didn't have this additional wrapper?

Reply:
we could rename `libp2pNode` to `libP2PWrapper`

Reply:
Addressed this comment in: https://github.com/onflow/flow-go/pull/195/commits/f13183572c705d1597947125f1b6a06af28ae530

Addressed factory comment in: https://github.com/onflow/flow-go/pull/195/commits/aa1e5c83bae3a2ec5b4336df87a64418d7b1937b